server/service/src/document/topological_sort.rs

```diff
@@ -0,0 +1,314 @@
+use std::collections::{HashMap, VecDeque};
+
+use repository::Document;
+
+pub trait TopoSortable {
+    fn id(&self) -> &String;
+    fn parents(&self) -> &Vec<String>;
+}
+
+impl TopoSortable for Document {
+    fn id(&self) -> &String {
+        &self.id
+    }
+
+    fn parents(&self) -> &Vec<String> {
+        &self.parents
+    }
+}
+
+/// Only returns items that are reachable from head
```

Comment:
Does that mean we only show history for changes that are done in current store ?

Reply:
yes that was the initial idea but as discussed it would be much better to show all known changes